kubernetes/schema_metadata.go

```diff
@@ -7,6 +7,14 @@ import (
 )
 
 func metadataFields(objectName string) map[string]*schema.Schema {
+	var objectNameRBAC = []string{"roleBinding", "clusterRoleBinding", "clusterRole", "role"}
```

Comment:
Apparently, there are other exceptions which require different validation rules, e.g.:

* CronJob: `DNS1035LabelMaxLength-11` https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/batch/validation/validation.go#L172

* ServiceAccount: `NameIsDNSLabel` https://github.com/kubernetes/kubernetes/blob/3a17d433aa7835de7eb69148c3c61b4f37c370d2/staging/src/k8s.io/apimachinery/pkg/api/validation/generic.go#L67

* Service: `NameIsDNS1035Label` https://github.com/kubernetes/kubernetes/blob/d0b0c0ae4591a83ab570dcad42982dfdaa14d04f/pkg/apis/core/validation/validation.go#L206

WDYT about introducing the validation function to use as new parameter to the `metadataFields` function?